### PR TITLE
Add an option to run Job in Density/Load config

### PR DESCRIPTION
**Pull request by gmarek**
cc @timothysc @jeremyeder 

@erictune @soltysh - I run this test and it seems to me that Job has noticeably worse performance than Deployment. I'll create an issue for this, but this PR is for easy repro.

### CODE REVIEW

**Comment by wojtek-t** [comment]
@gmarek - since daemons have already been merged, can you please rebase?

File: test/e2e/density.go [hunk 1]
```diff
@@ -252,13 +259,13 @@ func cleanupDensityTest(dtc DensityTestConfig) {
 		namespace := dtc.Configs[i].GetNamespace()
 		kind := dtc.Configs[i].GetKind()
 		// TODO: Remove Deployment guard once GC is implemented for Deployments.
-		if framework.TestContext.GarbageCollectorEnabled && kind != extensions.Kind("Deployment") {
+		if framework.TestContext.GarbageCollectorEnabled && kind != extensions.Kind("Deployment") && kind != batch.Kind("Job") {
```
Comment (LasTshaMAN) [inline test/e2e/density.go]: Couldn't this condition be exported to a separate function with a meaningful name ?

File: test/e2e/density.go [hunk 2]
```diff
@@ -258,7 +259,7 @@ func cleanupDensityTest(dtc DensityTestConfig) {
 		namespace := dtc.Configs[i].GetNamespace()
 		kind := dtc.Configs[i].GetKind()
 		// TODO: Remove Deployment guard once GC is implemented for Deployments.
-		if framework.TestContext.GarbageCollectorEnabled && kind != extensions.Kind("Deployment") {
+		if framework.TestContext.GarbageCollectorEnabled && kindSupportsGarbageCollector(kind) {
```
Comment (wojtek-t) [inline test/e2e/density.go]: Please remove TODO above.

File: test/utils/runners.go [hunk 3]
```diff
@@ -327,6 +333,66 @@ func (config *ReplicaSetConfig) create() error {
...
+	return batchinternal.Kind("Job")
+}
+
+func (config *JobConfig) create() error {
+	job := &batch.Job{
+		ObjectMeta: v1.ObjectMeta{
+			Name: config.Name,
+		},
+		Spec: batch.JobSpec{
+			Parallelism: func(i int) *int32 { x := int32(i); return &x }(config.Replicas),
```
Comment (wojtek-t) [inline test/utils/runners.go]: It's sad that we need to do such hacks :)

**Review by wojtek-t — COMMENTED** [review]
One very minor comment - other than that LGTM.

**Comment by gmarek** [comment]
Removed TODO. Applying label.

File: test/utils/runners.go (hunk 3 above)
Comment (soltysh) [inline test/utils/runners.go]: Sigh...

**Pull request merged**